checker/src/org/checkerframework/checker/formatter/FormatterAnnotatedTypeFactory.java

```diff
@@ -114,14 +114,18 @@ public FormatterQualifierHierarchy(MultiGraphFactory f) {
         @Override
         public boolean isSubtype(AnnotationMirror rhs, AnnotationMirror lhs) {
             if (AnnotationUtils.areSameIgnoringValues(rhs, FORMAT) &&
-                AnnotationUtils.areSameIgnoringValues(lhs, FORMAT))
-            {
+                AnnotationUtils.areSameIgnoringValues(lhs, FORMAT)) {
                 ConversionCategory[] rhsArgTypes =
                         treeUtil.formatAnnotationToCategories(rhs);
                 ConversionCategory[] lhsArgTypes =
                         treeUtil.formatAnnotationToCategories(lhs);
 
-                if (rhsArgTypes.length != lhsArgTypes.length) {
+                if (rhsArgTypes == null && lhsArgTypes == null) {
```

Comment:
I changed formatAnnotationToCategories to return null below, so these cases are necessary to handle that.